src/websockets.ml

```diff
@@ -21,8 +21,23 @@ module Wsprotocol (IO : Iteratees.Monad) = struct
 
   type 'a t = 'a I.t 
 
-  let base64encode s = modify Base64.encode s
-  let base64decode s = modify Base64.decode s
+  let sanitize s =
+    (* ignore control characters: see RFC4648.1 and RFC4648.3
+     * https://tools.ietf.org/html/rfc4648#section-3
+     * Note: \t = \009, \n = \012, \r = \015, \s = \032 *)
```

Comment:
I take that to mean that when encountering characters outside the base alphabet, the standard base64 behaviour is to reject the data.

But if you're writing a higher-level specification which _uses_ base64, then in that specification you're allowed to state explicitly some other behaviour, overriding the underlying base64 behaviour.